cpp/src/strings/search/find.cu

```diff
@@ -324,11 +333,11 @@ std::unique_ptr<column> contains_warp_parallel(strings_column_view const& input,
 
   if (!d_target.empty()) {
     // launch warp per string
-    auto d_strings = column_device_view::create(input.parent(), stream);
-    thrust::for_each_n(rmm::exec_policy(stream),
-                       thrust::make_counting_iterator<std::size_t>(0),
-                       static_cast<std::size_t>(input.size()) * cudf::detail::warp_size,
-                       contains_warp_fn{*d_strings, d_target, results_view.data<bool>()});
+    auto const d_strings     = column_device_view::create(input.parent(), stream);
+    constexpr int block_size = 256;
+    cudf::detail::grid_1d grid{input.size() * cudf::detail::warp_size, block_size};
```

Comment:
What if `input.size() * cudf::detail::warp_size` overflow? `grid_1d` only has `int` members.

Reply:
Oh I just found the similar code in other files (`attributes.cu` and `find.cu`). So this may be a new potential issue.

Reply:
You missed this line of code perhaps?
https://github.com/rapidsai/cudf/pull/13578/files#diff-048f86c21559b14f64f86aaeaa57776d366c3a4948a5aba7c0ab1a3801be87bcR292
```
 if (idx >= (d_strings.size() * cudf::detail::warp_size)) { return; }
```

Reply:
Wait, that line is inside the kernel, while this line is before kernel. If we have overflow here, we may still launch a kernel with some (large?) input. We should avoid launching the kernel from here.

Reply:
An overflow cannot technically occur here since this code path is only for long strings which is always much greater than 32 bytes on average. This means the number of rows * 32 will never overflow under these conditions.